Sources/BraintreePayPal/BTPayPalVaultRequest.swift

```diff
@@ -38,12 +38,12 @@ import BraintreeCore
             baseParameters["payer_email"] = userAuthenticationEmail
         }
         
-        if enablePayPalAppSwitch {
+        if enablePayPalAppSwitch, let universalLinkURL = BTAppContextSwitcher.sharedInstance.universalLink {
```

Comment:
Do you think we should return an error if they try to opt into `enablePayPalAppSwitch` but don't have a universal link set?

Reply:
🤔 that's a good question - I think it depends on the merchant experience we want. If this is not set we would get `nil` back as the `paypalAppApprovalUrl` so would just continue with the ASWeb based flow, but nothing would blow up. On the other side of things I can see a merchant ``_not_ wanting the ASWeb flow if they have set `enablePayPalAppSwitch` in which case we should add a check on the client that if true this value is also set. Which way are you leaning?

Reply:
I lean toward returning an error if they don't have the universal link set.

Actually brings me to another thought - we don't _need_ them to pass us the URL in their app delegate, do we? We shouldn't need it that early. We only need it at the time of making the GQ hermes POST request. We _could_ enforce this at the `BTPayPalRequest` init level. We can add an additional init specific to this flow that takes in everything we need: email, flag, link

Reply:
Yeah, that is a great question - I was just rolling with the design we had outlined. Technically I don't think the `returnURLScheme` for the Venmo flow needs to be set in `BTAppContextSwitcher` either. It's also not used until the `tokenize` method is called. Do you know historically why those were set in that manner instead of on the clients?

I am down to move this into a new init - I think that makes sense in this PR. We would also need to make a very fast follow docs PR since that was merged in today to be released Wednesday. I think if we make it required in the init we won't need to throw an error from the request or do a check in the client and throw the error there.

If that sounds good to you I can make the change here and get up a docs update as well?

Reply:
@scannillo this update is reflected in the PR - if we are good with this change, I will open a new docs PR and ping the docs team as well as update our HLD. In `BTPayPalVaultRequest` I added a new `init` with the required properties and made `universalLink` an internal var on the class so it can only be set on the `init` - this will enforce it as required without us needing to throw an error in the `init` for this flow.

Reply:
> Do you know historically why those were set in that manner instead of on the clients?

Unfortunately I don't know why it was that way

Reply:
No worries, it's a great question in any case! I'm going to add a `NEXT_MAJOR_VERSION` note to `returnURLScheme` so we can move that to the clients in the future. 🚀